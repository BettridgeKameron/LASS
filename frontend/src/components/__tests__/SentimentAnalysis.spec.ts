import { describe, it, expect, vi } from 'vitest';
import SentAnalyze from '@/views/SentimentAnalysis.vue';
import { mount } from '@vue/test-utils';

describe('SentimentAnalysis.vue', () => {
  it('should call the analyzeSentiment method when the button is clicked', async () => {
    const wrapper = mount(SentAnalyze);

    // Find the button element
    const button = wrapper.find('button');

    // Mock the analyzeSentiment method (optional)
    const mockAnalyzeSentiment = vi.fn();
    wrapper.vm.analyzeSentiment = mockAnalyzeSentiment; // Replace with actual method if needed

    // Trigger the click event on the button
    await button.trigger('click');

    // Assert that the analyzeSentiment method was called
    expect(mockAnalyzeSentiment).toHaveBeenCalled();
  });
});