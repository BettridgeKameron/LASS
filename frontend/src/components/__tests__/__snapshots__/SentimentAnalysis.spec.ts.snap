// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sentiment Analysis > should render correctly 1`] = `
"<div class="container mx-auto p-4 max-w-5xl">
  <h1 class="text-4xl font-bold mb-2 py-5">Sentiment Analysis</h1>
  <div class="flex flex-col space-y-4"><textarea style="height: 500px; resize: none;" class="w-full border-2 border-gray-300 p-3 rounded-md" placeholder="Type something..." rows="4"></textarea><button class="w-full bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"> Send </button>
    <div class="p-6 rounded shadow bg-neutral-content border-2 border-gray-300" style="word-break: break-all;">
      <div class="text-neutral" style="color: rgba(0, 0, 0, 0.5);">Sentiment Output</div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`Should make sure page is working > should render correctly 1`] = `
"<div class="container mx-auto p-4 max-w-5xl">
  <h1 class="text-4xl font-bold mb-2 py-5">Sentiment Analysis</h1>
  <div class="flex flex-col space-y-4"><textarea style="height: 500px; resize: none;" class="w-full border-2 border-gray-300 p-3 rounded-md" placeholder="Type something..." rows="4"></textarea><button class="w-full bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"> Send </button>
    <div class="p-6 rounded shadow bg-neutral-content border-2 border-gray-300" style="word-break: break-all;">
      <div class="text-neutral" style="color: rgba(0, 0, 0, 0.5);">Sentiment Output</div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
